src/modules/uORB/uORB_tests/uORBTest_UnitTest.cpp

```diff
@@ -582,6 +588,148 @@ int uORBTest::UnitTest::test_multi_reversed()
 	return test_note("PASS multi-topic reversed");
 }
 
+int uORBTest::UnitTest::test_wrap_around()
+{
+	test_note("Testing orb wrap-around");
+
+	orb_test_medium_s t{};
+	orb_test_medium_s u{};
+	orb_advert_t ptopic{nullptr};
+	bool updated{false};
+
+	int sfd = orb_subscribe(ORB_ID(orb_test_medium_wrap_around));
+
+	if (sfd < 0) {
+		return test_fail("subscribe failed: %d", errno);
+	}
+
+	const int queue_size = 16;
+	t.val = 0;
+	ptopic = orb_advertise_queue(ORB_ID(orb_test_medium_wrap_around), &t, queue_size);
+
+	if (ptopic == nullptr) {
+		return test_fail("advertise failed: %d", errno);
+	}
+
+	// Set generation to the location where wrap-around is about to be
+	{
+		auto node = uORB::Manager::get_instance()->get_device_master()->getDeviceNode(ORB_ID(orb_test_medium_wrap_around), 0);
+
+		if (node == nullptr) {
+			return test_fail("get device node failed.");
+		}
+
+		// Set to (0 - queue_size / 2) => (UINT_MAX - queue_size / 2 + 1)
+		set_generation(*node, unsigned(-(queue_size / 2)));
+
+		if (node->updates_available(0) != unsigned(-(queue_size / 2))) {
+			return test_fail("set generation failed.");
+		}
+
+		// Refresh the subscriber's generation
```

Comment:
Can you create the subscriber after setting the generation instead? It would reflect the typical use-case better.

Reply:
It's done.